torch/utils/data/dataset.py

```diff
@@ -35,3 +35,51 @@ def __getitem__(self, index):
 
     def __len__(self):
         return self.data_tensor.size(0)
+
+
+class ConcatDataset(Dataset):
+    """
+    Dataset to concatenate multiple datasets.
+    Purpose: useful to assemble different existing datasets, possibly
+    large-scale datasets as the concatenation operation is done in an
+    on-the-fly manner.
+    Args:
+        datasets (iterable): List of datasets to be concatenated
+    """
+
+    @staticmethod
+    def cumsum(sequence):
```

Comment:
You still can do something like `torch.cumsum(torch.LongTensor([1, 2, 3]))`, but I suppose it's not super important in here as it's only used in the constructor.